static/js/components/property/smoke.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * SmokeProperty
+ *
+ * A bubble showing a smoke label.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+'use strict';
+
+const BaseComponent = require('../base-component');
+const fluent = require('../../fluent');
+
+class SmokeProperty extends BaseComponent {
+  constructor() {
+    const template = document.createElement('template');
+    template.innerHTML = `
+  <style>
+    :host {
+      display: inline-block;
+      contain: content;
+      text-align: center;
+      color: white;
+      font-size: 1.6rem;
+    }
+
+    .webthing-smoke-property-container {
+      width: 10rem;
+      height: 10rem;
+      border-radius: 5rem;
+      border: 0.2rem solid white;
+      background-color: #89b6d6;
+      position: relative;
+    }
+
+    .webthing-smoke-property-container.inverted {
+      color: white;
+      background-color: #ff5555;
+      border: 0.2rem solid #ff5555;
+    }
+
+    .webthing-smoke-property-value {
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      font-weight: bold;
+      width: 100%;
+      font-size: 1.4rem;
+      text-transform: uppercase;
+    }
+
+    .webthing-smoke-property-container.inverted
+    .webthing-smoke-property-value {
+      top: calc(50% + 1rem);
+    }
+
+    .webthing-smoke-property-image {
+      display: none;
+    }
+
+    .webthing-smoke-property-container.inverted
+    .webthing-smoke-property-image {
+      display: block;
+      width: 3.2rem;
+      position: absolute;
+      left: calc(50% - 1.6rem);
+      top: 1.6rem;
```

Comment:
It would be great if this icon (and its label) could move down slightly too so it looks more vertically centered in the "smoke" state as in the mockups.

Reply:
Done.